CMakeLists.txt

```diff
@@ -112,21 +114,20 @@ target_include_directories(${IRODS_S3_API_PERSISTENCE_PLUGIN} PRIVATE ${IRODS_EX
 
 add_subdirectory(src)
 
+install(
+  TARGETS
+  ${IRODS_S3_API_BINARY_NAME}
+  ${IRODS_S3_API_AUTH_RESOLVER_NAME}
+  ${IRODS_S3_API_BUCKET_RESOLVER_PLUGIN_NAME}
+  DESTINATION "${LIBEXECDIR}/irods/clients/s3_api")
```

Comment:
This looks like a new convention.

Does `${LIBEXECDIR}` expand to something other than `libexec`? What are the pros of using this?

Are you envisioning a new layout for where we install plugins, executables, etc?

Reply:
> As far as I'm aware, we do not (as far as I'm aware) currently have a convention for where to put clients we don't want in $PATH ...

Wait, why don't we want this in $PATH? `/usr/local/bin` or `/usr/bin` seem like the appropriate place for the executable.

`/usr/lib` or the platform equivalent is fine for the shared libraries.

Reply:
> Wait, why don't we want the executable in $PATH?

Probably because of the current plugin architecture. I'd be fine with putting it in `$PATH` once the plugins can live in a separate directory.

> `/usr/local/bin` or `/usr/bin` seem like the appropriate place for the executable.

As a general rule, you don't want packages installing anything into `/usr/local`

Reply:
> Probably because of the current plugin architecture. I'd be fine with putting it in $PATH once the plugins can live in a separate directory.

I see. So we need to define where plugins for this app go and update the app to look there. Hence #81.

> As a general rule, you don't want packages installing anything into `/usr/local`

I think I can guess why, but please explain for future readers and I.

Sounds like `/usr/bin` would be the answer, but the plugin loading blocks us from doing that.

Reply:
> > As a general rule, you don't want packages installing anything into `/usr/local`
> 
> I think I can guess why, but please explain for future readers and I.

The simple version is that `/usr/local` is for software built locally and untracked by the package manager.

> Sounds like `/usr/bin` would be the answer, but the plugin loading blocks us from doing that.

My thoughts as well. In the meantime, I think `${LIBDIR}/irods/clients/s3_api` is a good temporary location. I'll update the relevant commit.

Reply:
Sounds good.

We'll get plugin directory issue resolved before the next release.

When ready, please mark this conversation as resolved.